src/lib/scripts/verseSelectUtil.ts

```diff
@@ -0,0 +1,105 @@
+import { selectedVerses } from '$lib/data/stores';
+import config from '$lib/data/config';
+
+export type SelectedReference = {
+    docSet: string;
+    book: string;
+    chapter: string;
+    verse: string;
+};
+export function onClickText(e) {
+    let target = e.target;
+
+    while (!isSelectableText(target) && !isClickable(target) && !isMain(target)) {
+        target = target.parentNode;
+    }
+    if (isSelectableText(target)) {
+        const id = removeIdSuffixes(target.id);
+        if (target.className.indexOf('selected') < 0) {
+            const maxSelections = config.mainFeatures['annotation-max-select'];
+            const currentLength = selectedVerses.length();
+            if (currentLength < maxSelections) {
+                modifyClassOfElements(id, 'selected', true);
+                selectedVerses.addVerse(id);
+            }
+        } else {
+            modifyClassOfElements(id, 'selected', false);
+            selectedVerses.removeVerse(id);
+        }
+    }
+}
+// Deselect all elements
+export function deselectAllElements() {
+    const els = document.getElementsByTagName('div');
+    for (let i = 0; i < els.length; i++) {
+        if (els[i].id != '') {
+            els[i].className = els[i].className.replace('selected', '');
```

Comment:
This can be:
`els[i].classList.remove('selected');`

Reply:
Done